src/App.tsx

```diff
@@ -15,36 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SORT {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+function sort(goods: string[], sortField: SORT, isReversed: boolean): string[] {
+  const sortedGoods = [...goods];
+
+  switch (sortField) {
+    case SORT.ALPHABET:
+      sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SORT.LENGTH:
+      sortedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    case SORT.DEFAULT:
+    default:
+      break;
+  }
+
+  return isReversed ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SORT>(SORT.DEFAULT);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const newGoodsFromServer = sort(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': SORT.ALPHABET !== sortField,
+          })}
+          onClick={() => {
+            setSortField(SORT.ALPHABET);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': SORT.LENGTH !== sortField,
+          })}
+          onClick={() => {
+            setSortField(SORT.LENGTH);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => {
+            setIsReversed(prev => !prev);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(SORT.DEFAULT);
+              setIsReversed(false);
+            }}
```

Comment:
Move this logic from jsx to the helper function and use it here
```suggestion
            onClick={reset}
```